client/src/components/base/ListTable/index.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+type test = {
+  [key: string]: number | {c: React.ReactNode, colSpan: number}
+}
+
+type ListTableProps = {
+  checkable: boolean;
+  header: Array<{ id: string; name: string; width?: string; rowSpan?: number }>;
+  body: test [];
+  selectedItems?: Set<number>;
+  onCheck?: (id: number) => void;
+  onCheckAll?: (e) => void
+};
+
+const ListTable = ({ checkable, header, body, selectedItems ,onCheck , onCheckAll}: ListTableProps) => {
+  const calculateThWidth = () => {
+    return header.reduce((acc, { width }) => [...acc, width ? width : '10%'], []);
+  };
```

Comment:
```suggestion
const calculatedTheadWitdh = () => {
    return header.map(({width}) => (!!width ? width : '10%'));
}
또는
const calculatedTheadWidth = header.map(({width}) => (!!width ? width : '10%'));
```


reduce는 값을 하나로 합치는 의미로 사용되는 경우가 많은걸로 아는데 map을 사용해도 괜찮을 듯 하네요!

그리고 함수로 만들지 않고 변수로 만들어서 사용해도 괜찮을 거 같아요!

Reply:
오 그렇네요 변수로 만드는게 나을듯 함니다